src/NHibernate/Impl/SessionFactoryImpl.cs

```diff
@@ -248,7 +248,10 @@ public SessionFactoryImpl(Configuration cfg, IMapping mapping, Settings settings
 					if (cache != null)
 					{
 						caches.Add(cacheRegion, cache);
-						((IDictionary<string, ICache>)allCacheRegions).Add(cache.RegionName, cache.Cache);
+						if (!allCacheRegions.TryAdd(cache.RegionName, cache.Cache))
+						{
+							throw new HibernateException("Key already existed");
```

Comment:
As per guidelines, exception messages are supposed to be proper sentences, which includes a period at the end. (Multiple locations.)